WrapImage/nifti_wrapper.py

```diff
@@ -0,0 +1,64 @@
+import argparse
+import json
+import os
+import nibabel as nib
+import numpy as np
+
+def read_nifti_4d(input_file):
+    """
+    For reading the 4d nifti image
+    """
+    nifti_img = nib.load(input_file)
+    return nifti_img.get_fdata()
+
+def read_json_file(json_file):
+    """
+    For reading the json file
+    """
+
+    if not os.path.exists(json_file):
+        raise FileNotFoundError(f"File '{json_file}' not found.")
+
+    with open(json_file, "r") as f:
+        try:
+            json_data = json.load(f)
+        except json.JSONDecodeError as e:
+            raise ValueError(f"Error decoding JSON in file '{json_file}': {e}")
+
+    return json_data
+
+def save_nifti_3d(data, output_file):
+    """
+    For saving the 3d nifti images of the output of the algorithm
+    """
+    output_img = nib.Nifti1Image(data, np.eye(4))
```

Comment:
Creating this output image might necessitate copying aspects of the input image. This function should probably accept or at least pass through arguments to `Nifti1Image`. https://nipy.org/nibabel/reference/nibabel.nifti1.html#nibabel.nifti1.Nifti1Image

Reasonable defaults for whatever n-dimensional image make sense though.